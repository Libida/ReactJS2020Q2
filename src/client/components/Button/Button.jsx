import React from 'react';
import PropTypes from 'prop-types';
import './Button.scss';

function Button({type, text, incomeClasses, isDisabled}) {
    return (
        <button type={type} className={`btn ${incomeClasses}`} disabled={isDisabled}>{text}</button>
    );
}

Button.propsType = {
    type: PropTypes.string,
    text: PropTypes.string,
    incomeClasses: PropTypes.string,
    isDisabled: PropTypes.bool,
};

Button.defaultProps = {
    type: 'button',
    incomeClasses: '',
};

export default React.memo(Button);
