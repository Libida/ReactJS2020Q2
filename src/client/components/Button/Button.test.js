import React from 'react';
import renderer from 'react-test-renderer';
import {Button} from './Button';

describe('Button component', () => {
    const component = renderer.create(
        <Button text="Search" />
    );

    let tree = component.toJSON();

    it('should be rendered properly with default props', () => {
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered properly if disabled', () => {
        tree.props.isDisabled = 'true';
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered properly with different type', () => {
        tree.props.type = 'submit';
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered properly with additional classes', () => {
        tree.props.incomeClasses = 'btn-primary';
        expect(tree).toMatchSnapshot();
    });
});
