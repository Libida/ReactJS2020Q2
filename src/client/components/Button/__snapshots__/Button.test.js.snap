// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should be rendered properly if disabled 1`] = `
<button
  className="btn "
  isDisabled="true"
  type="button"
>
  Search
</button>
`;

exports[`Button component should be rendered properly with additional classes 1`] = `
<button
  className="btn "
  incomeClasses="btn-primary"
  isDisabled="true"
  type="submit"
>
  Search
</button>
`;

exports[`Button component should be rendered properly with default props 1`] = `
<button
  className="btn "
  type="button"
>
  Search
</button>
`;

exports[`Button component should be rendered properly with different type 1`] = `
<button
  className="btn "
  isDisabled="true"
  type="submit"
>
  Search
</button>
`;
